### PR TITLE
Changes for `lightning_payjoin` integration

### PR DESCRIPTION
**Pull request by jbesraa**
Here are few commits I have while working on the payjoin integration with `ldk_node`  https://github.com/lightningdevkit/ldk-node/pull/257.

### CODE REVIEW

**Comment by DanGould** [comment]
All of these changes are new API surface. I imagine some of these will be required but others might be able to be worked around. May you please enumerate *why* each change is required or is a better design? I'd especially like to better understand the changes that affect type/method visibility.

I think #69 might be better to fix with a dedicated PR designed to fix the problem in general, allowing for substitution of the original payment output with any number of outputs, rather than an additional substitution method that increases the surface area of the API. Glad you're having success finding compatiblity with lightning payjoin integration in any case.

File: payjoin/src/lib.rs [hunk 1]
```diff
@@ -47,5 +47,5 @@ pub(crate) mod weight;
 
 #[cfg(feature = "base64")]
 pub use bitcoin::base64;
-pub use uri::{PjParseError, PjUri, PjUriBuilder, Uri};
+pub use uri::{MaybePayjoinExtras, PjParseError, PjUri, PjUriBuilder, Uri};
```
Comment (jbesraa) [inline payjoin/src/lib.rs:50]: this is useful in testing where we want to mutate the pjuri in order to change test direction(ie start with a pj uri and then maybe change it to something different) example: https://github.com/lightningdevkit/ldk-node/pull/257/files#diff-ebba69f49eedcd3dd6a656eff964c6184b6d29c02fbad6a49d3b41c87cc3de79R192

File: payjoin/src/uri.rs [hunk 2]
```diff
@@ -24,7 +24,7 @@ impl MaybePayjoinExtras {
 
 #[derive(Clone)]
 pub struct PayjoinExtras {
-    pub(crate) endpoint: Url,
+    pub endpoint: Url,
```
Comment (jbesraa) [inline payjoin/src/uri.rs:27]: this is useful in testing where we want to mutate the pjuri in order to change test direction(ie start with a pj uri and then maybe change it to something different) example: https://github.com/lightningdevkit/ldk-node/pull/257/files#diff-ebba69f49eedcd3dd6a656eff964c6184b6d29c02fbad6a49d3b41c87cc3de79R192

File: payjoin/src/receive/mod.rs [hunk 3]
```diff
@@ -481,6 +481,16 @@ impl ProvisionalProposal {
         self.payjoin_psbt.unsigned_tx.output[self.owned_vouts[0]].script_pubkey = script;
     }
 
+    /// payjoin request value in satoshis
```
Comment (jbesraa) [inline payjoin/src/receive/mod.rs:484]: This is useful in lightning context in order to get a scheduled channel based on the payjoin amount received
exmple: https://github.com/lightningdevkit/ldk-node/pull/257/files#diff-d07cd679b2d6743b550aec08c91b33e92bdd40a9556385925c987e9d82b222c5R89

**Comment by DanGould** [comment]
You should be able to substitute the output with *many* outputs at random index, not just one. I'm not exactly sure what that interface looks like yet.

**Comment by jbesraa** [comment]
Took out the `substitute_output_script` commits

**Pull request closed without merging**